### PR TITLE
Tellie dq

### PR DESCRIPTION
**Pull request by ChMills**
Made the tellie DQ page load faster using a similar method to #170. Also updated the physics DQ page for individual runs with a couple of extra parameters and removed a few obsolete ones.

### CODE REVIEW

File: minard/HLDQTools.py [hunk 1]
```diff
@@ -52,68 +52,86 @@ def generateHLDQProcStatus(ratdbDict):
...
     # select all runs which have the external source and tellie bits checked
     result = conn.execute("SELECT run FROM run_state WHERE (run_type & 2064) = 2064 ORDER BY run DESC LIMIT %s OFFSET %s", (limit,offset))
     return [row[0] for row in result.fetchall()]
 
-def import_TELLIEDQ_ratdb(runNumber):
+def import_TELLIEDQ_ratdb(runs):
+    if type(runs) == int:
+        runs = [runs]
+    else:
+        pass
```
Comment (tlatorre-uchicago) [inline minard/HLDQTools.py]: This else block seems redundant.

File: minard/HLDQTools.py [hunk 2]
```diff
@@ -52,68 +52,86 @@ def generateHLDQProcStatus(ratdbDict):
...
-    runInformation["max_nhit_check_subruns"] = data["check_params"]["max_nhit_check"]
-    runInformation["peak_number_check_subruns"] = data["check_params"]["peak_number_check"]
-    runInformation["prompt_peak_amplitude_check_subruns"] = data["check_params"]["prompt_peak_amplitude_check"]
-    runInformation["prompt_peak_adc_count_check_subruns"] = data["check_params"]["prompt_peak_adc_count_check"]
-    runInformation["adc_peak_time_spacing_check_subruns"] = data["check_params"]["adc_peak_time_spacing_check"]
-    runInformation["pulse_delay_efficiency_check_subruns"] = data["check_params"]["pulse_delay_efficiency_check"]
-    runInformation["subrun_run_length_check"] = data["check_params"]["subrun_run_length_check"]
-    runInformation["correct_fibre_check_subruns"] = data["check_params"]["correct_fibre_check"]
-    runInformation["trigger_check_subruns"] = data["check_params"]["trigger_check"]
+    if data == {}:
```
Comment (tlatorre-uchicago) [inline minard/HLDQTools.py]: This probably works, but I think you can achieve the same (and a little easier to read) by doing:

```python
if data:
    return runs, -1, -1
```

or

```python
if len(data) == 0:
    return runs, -1, -1
```

File: minard/views.py [hunk 3]
```diff
@@ -1305,19 +1305,23 @@ def calibdq_tellie():
...
     limit = request.args.get("limit", 10, type=int)
     offset = request.args.get("offset", 0, type=int)
-    run_numbers = HLDQTools.import_TELLIE_runnumbers(limit=limit, offset=offset)
-    for num in run_numbers:
-        run_num, check_params, runInformation = HLDQTools.import_TELLIEDQ_ratdb(num)
-        run_dict[num] = check_params
+    # run_numbers = HLDQTools.import_TELLIE_runnumbers(limit=limit, offset=offset)
+    # for num in run_numbers:
+    #     run_num, check_params, runInformation = HLDQTools.import_TELLIEDQ_ratdb(num)
+    #     run_dict[num] = check_params
```
Comment (tlatorre-uchicago) [inline minard/views.py]: Can these lines be deleted?

File: minard/templates/calibdq_tellie.html [hunk 4]
```diff
@@ -14,6 +14,13 @@ <h1>TELLIE Data Quality</h1>
     <a class="btn btn-outline-primary btn-lg" href="{{ url_for("calibdq_tellie", limit=limit, offset=offset-limit) }}">Back</a>
     <a class="btn btn-outline-primary btn-lg" href="{{ url_for("calibdq_tellie", limit=limit, offset=offset+limit) }}">Next</a>
     {% endif %}
+    Runs per page:
+    <select onchange="location=this.options[this.selectedIndex].value;">
+        <option value="{{url_for("calibdq_tellie", limit=10, offset=offset)}}">10</option>
+        <option value="{{url_for("calibdq_tellie", limit=25, offset=offset)}}">25</option>
+        <option value="{{url_for("calibdq_tellie", limit=50, offset=offset)}}">50</option>
+        <option value="{{url_for("calibdq_tellie", limit=100, offset=offset)}}">100</option>
+    </select>
```
Comment (tlatorre-uchicago) [inline minard/templates/calibdq_tellie.html]: Currently if you select a different limit the page loads but the drop down menu always displays 10. You can have a look at the stream page code here: https://github.com/snoplus/minard/blob/master/minard/templates/snostream.html#L28 to see how to automatically have the dorrect limit selected.

**Comment by ChMills** [comment]
@tlatorre-uchicago I've made the changes that you've highlighted, should look a little better and the drop down menu should be fixed. For your general comments:
1. I have added this menu to both the physics and smellie dq pages.
2. Some runs do have information displayed, for example run 201606 - https://snopl.us/monitoring/calibdq_tellie/201606/. I'm unsure why some of the more recent runs do not have any information, a quick look at the runs page suggests that if they fail the run processor DQHL check then nothing gets displayed here.

**Comment by tlatorre-uchicago** [comment]
I just tested this out and got an error when visiting the TELLIE DQ page:

```python
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/raid/home/tlatorre/minard/minard/__init__.py", line 34, in __call__
    return self.app(environ, start_response)
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/lib/python2.6/site-packages/Flask-0.10-py2.6.egg/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/raid/home/tlatorre/minard/minard/views.py", line 1311, in calibdq_tellie
    run_dict[num] = check_params[num]
TypeError: 'int' object is unsubscriptable
```

**Comment by ChMills** [comment]
@tlatorre-uchicago Good catch, error occurred when none of the runs that are displayed have DQ tables. I have fixed this.

**Pull request merged**